ci: fix documentation generation

We have an error in workflow run for documentation generation:

```
Run bash ./build_mkdocs.sh
Docs saved to: doc/references/cli.md
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /home/runner/work/robotoff/robotoff/gh_pages
WARNING  -  Documentation file 'introduction/contributing.md' contains a link to '../robotoff/app/api.py' which is not found in the documentation files.
WARNING  -  Documentation file 'introduction/contributing.md' contains a link to '../robotoff/settings.py' which is not found in the documentation files.
Aborted with 2 warnings in strict mode!
Error: Process completed with exit code 1.
```

1. Fix the files, 
1. Also fix the Makefile to also put documentation generation in strict mode, so that `make checks` should catch this (currently it was not failing)

--- a/doc/introduction/contributing.md
+++ b/doc/introduction/contributing.md
@@ -76,7 +76,7 @@ Ready to contribute code? Here's how to set up Robotoff for local development.
 
     You can make a post request through [Postman](https://www.postman.com/) or simply paste the url in a web browser to make a get request like this one http://localhost:5500/api/v1/insights/
 
-    The mapping of functions and API path is [here](../../robotoff/app/api.py#L1083) 
+    The mapping of functions and API path is at the end of robotoff/app/api.py
 
     If you are on a local install:
 
@@ -88,7 +88,7 @@ Ready to contribute code? Here's how to set up Robotoff for local development.
     poetry run pytest tests
     ```    
 
-    Before running the test cases make sure you have a database created. Have a look at [.env](../../robotoff/settings.py#L131) the default database name, user, and password is
+    Before running the test cases make sure you have a database created. Have a look at .env and robotoff/settings.py the default database name, user, and password is
 
     ```
     postgres

